### PR TITLE
feat: normalise # in URLs for referrer classification

### PR DESCRIPTION
**Pull request by asthabh23**
## Description
This PR handles normalizing `#` in URLs for referrer classification 

## Related Issue
#31 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by langswei** [comment]
Do we need to remove `/#` or just `#` to normalize?

**Comment by asthabh23** [comment]
> Do we need to remove `/#` or just `#` to normalize?

I wonder the same, I did `/#` initially.. but then i think `/` is still needed?

**Comment by langswei** [comment]
My first thought is that consolidating `https://www.google.com` and `https://www.google.com/#` means we need to remove two chars: `/#`.  But I suppose this needs to be tested because I'm not sure if the redundancy in the facet was driven by UI vs. backend.

**Comment by asthabh23** [comment]
indeed, I also think that just removing the chars might not help.. removing duplicates will still be needed maybe separately., but I wanted to test that as next step once I get to test this change.
(since till the time it's not merged here, I cant test in explorer)

File: facets.js [hunk 1]
```diff
@@ -154,20 +154,26 @@ export const facets = {
...
-        referrerClass.type,
-        `*:${referrerClass.vendor}`,
-      ] : source;
-    })
-    .flat(),
+  enterSource: (bundle) => {
+    const normalizedSources = bundle.events
+      .filter((evt) => evt.checkpoint === 'enter')
+      .map((evt) => evt.source)
+      .filter((source) => source)
```
Comment (trieloff) [inline facets.js:161]:
```suggestion
      .filter((source) => source)
      .map((source) => source.replace(/\/#$/, '')
```

that's how I'd do it.

**Review by trieloff — CHANGES_REQUESTED** [review]
Let's also include a test case for that specific scenario

File: facets.js (hunk 1 above)
Comment (asthabh23) [reply facets.js]: updated.

**Review by trieloff — APPROVED** [review]

**Pull request merged**